src/rtp/ext.rs

```diff
@@ -44,8 +52,50 @@ pub enum Extension {
     FrameMarking,
     /// <http://www.webrtc.org/experiments/rtp-hdrext/color-space>
     ColorSpace,
+
     /// Not recognized UR, but it could still be user parseable.
-    UnknownUri(String),
+    #[doc(hidden)]
+    UnknownUri(String, Arc<dyn ExtensionSerializer>),
+}
+
+// TODO: think this through. Is it unwind safe?
+impl UnwindSafe for Extension {}
+
+/// Trait for parsing/writing user RTP header extensions.
+pub trait ExtensionSerializer: Debug + Send + Sync + 'static {
```

Comment:
I think this needs a different name, `Serializer` implies only serializing, but this does both Serializing and Deserializing. Rust folks will have this understanding of those terms because of `serde`

Reply:
I don't have any, I [asked ChatGPT](https://chat.openai.com/share/b60cb0fb-cb34-46f4-b28b-2c6328ae8174) and it also didn't have any good ideas.

I guess we can keep it unless someone else has a good idea

Reply:
Alright. If ChatGPT says so, I'll do a force unwrap().